src/main.py

```diff
@@ -26,53 +26,65 @@ def main():
+                blocked_connection_timeout=settings.MQ_TIMEOUT
+            )
+        )
+
+        logger.info("init message queue channel")
+        channel = connection.channel()
 
-    connection = pika.BlockingConnection(
-        pika.ConnectionParameters(
-            host=settings.MQ_HOST,
-            heartbeat=settings.MQ_HEARTBEAT,
-            blocked_connection_timeout=settings.MQ_TIMEOUT
+        channel.exchange_declare(
+            exchange=settings.MQ_EXCHANGE, 
+            exchange_type='topic'
         )
-    )
 
-    logger.info("init message queue channel")
-    channel = connection.channel()
+        result = channel.queue_declare(settings.MQ_QUEUE, exclusive=False)
 
-    channel.exchange_declare(
-        exchange=settings.MQ_EXCHANGE, 
-        exchange_type='topic'
-    )
+        queue_name = result.method.queue
+    
+        logger.info("bind all keys to queue")
+        for binding_key in settings.MQ_BINDKEYS:
+            channel.queue_bind(
+                exchange=settings.MQ_EXCHANGE, 
+                queue=queue_name, 
+                routing_key=binding_key
+            )
 
-    result = channel.queue_declare(settings.MQ_QUEUE, exclusive=False)
+        # switch message round-robin assignment to ready processes first
+        # Force service to handle one message at the time!
+        channel.basic_qos(prefetch_count=1) 
 
-    queue_name = result.method.queue
-   
-    logger.info("bind all keys to queue")
-    for binding_key in settings.MQ_BINDKEYS:
-        channel.queue_bind(
-            exchange=settings.MQ_EXCHANGE, 
+        # register consuming function as callback
+        channel.basic_consume(
             queue=queue_name, 
-            routing_key=binding_key
+            on_message_callback=consume_handler
         )
 
-    # switch message round-robin assignment to ready processes first
-    # Force service to handle one message at the time!
-    channel.basic_qos(prefetch_count=1) 
-
-    # register consuming function as callback
-    channel.basic_consume(
-        queue=queue_name, 
-        on_message_callback=consume_handler
-    )
-
-    try:
-        logger.info("start consuming")
-        channel.start_consuming()
-    except KeyboardInterrupt:
-        logger.info("interactive termination")   
-        channel.stop_consuming()
-        connection.close()
+        try:
+            logger.info("start consuming")
+            channel.start_consuming()
+        except KeyboardInterrupt:
+            logger.info("interactive termination")   
+            channel.stop_consuming()
+            break
+        # Don't recover if connection was closed by broker
+        except pika.exceptions.ConnectionClosedByBroker:
+            break
+        # Don't recover on channel errors
+        except pika.exceptions.AMQPChannelError:
+            break
+        # Recover on all other connection errors
+        except pika.exceptions.AMQPConnectionError:
```

Comment:
As the last statement in a loop `continue` makes no sense. Please comment but leave them for documentation purposes.